Mathlib/FieldTheory/Finite/Basic.lean

```diff
@@ -201,6 +228,84 @@ theorem sum_pow_units [DecidableEq K] (i : ℕ) :
         apply forall_congr'; intro x; simp [Units.ext_iff]
 #align finite_field.sum_pow_units FiniteField.sum_pow_units
 
+/-- The sum of a nontrivial subgroup of the units of a field is zero. -/
+theorem sum_subgroup_units_zero_of_ne_bot {G : Subgroup Kˣ} [Fintype G] (hg : G ≠ ⊥) :
+    ∑ x : G, (x.val : K) = 0 := by
+  rw [Subgroup.ne_bot_iff_exists_ne_one] at hg
+  rcases hg with ⟨a, ha⟩
+  -- The action of a on G as an embedding
+  have hinj := mul_right_injective a
+  let a_mul_emb : G ↪ G := ⟨(a * ·), hinj⟩
+  -- let a_mul_emb : G ↪ G := by exact mulLeftEmbedding a -- TODO factor this? In library?
```

Comment:
you should be using `:= mulLeftEmbedding a` -- why aren't you?

Reply:
I had trouble swapping these out, If you want me to I can try harder.

Reply:
Ok I figured it out